File: src/org/bireme/dia/analysis/SimpleKeywordAnalyzer.java
package org.bireme.dia.analysis;

import org.apache.lucene.analysis.Analyzer;
import org.apache.lucene.analysis.TokenStream;
import org.apache.lucene.analysis.Tokenizer;
import org.apache.lucene.analysis.core.KeywordTokenizer;
import org.apache.lucene.analysis.core.LowerCaseFilter;
import org.apache.lucene.analysis.miscellaneous.ASCIIFoldingFilter;

/**
 * CharTokenizer limits token width to 255 characters, though.
 * This implementation assumes keywords are 255 in length or less.
 */
public class SimpleKeywordAnalyzer extends Analyzer {
    @Override
    protected TokenStreamComponents createComponents(final String fieldName) {
        final Tokenizer source = new KeywordTokenizer();
        final TokenStream filter1 = new ASCIIFoldingFilter(source);
        final TokenStream filter2 = new LowerCaseFilter(filter1);

        return new TokenStreamComponents(source, filter2);
    }
}

File: src/main/scala/org/bireme/dia/analysis/SimpleKeywordAnalyzer.scala
package org.bireme.dia.analysis

import org.apache.lucene.analysis.{Analyzer, LowerCaseFilter}
import org.apache.lucene.analysis.Analyzer.TokenStreamComponents
import org.apache.lucene.analysis.core.KeywordTokenizer
import org.apache.lucene.analysis.miscellaneous.ASCIIFoldingFilter

class SimpleKeywordAnalyzer extends Analyzer:
  override protected def createComponents(fieldName: String): TokenStreamComponents =
    val source = KeywordTokenizer()
    val filter1 = ASCIIFoldingFilter(source)
    val filter2 = LowerCaseFilter(filter1)

    new TokenStreamComponents(source, filter2)
